modules/apps/29-fee/keeper/escrow.go

```diff
@@ -168,9 +168,13 @@ func (k Keeper) distributeFee(ctx sdk.Context, receiver, refundAccAddress sdk.Ac
 			return // if sending to the refund address fails, no-op
 		}
 
-		EmitDistributeFeeEvent(ctx, refundAccAddress.String(), fee)
+		EmitDistributeFeeEvent(ctx, types.Refund, refundAccAddress.String(), fee)
 	} else {
-		EmitDistributeFeeEvent(ctx, receiver.String(), fee)
+		if bytes.Equal(receiver, refundAccAddress) {
```

Comment:
This check will take a reward for a refund when the same address is used to incentivize and to relay the packet. Not sure if it's a big issue, though.

Reply:
As an alternative to this check here, we could pass an extra argument to `distributeFee` indicating whether we are rewarding or refunding.